modules/internal/ArrayViewSlice.chpl

```diff
@@ -0,0 +1,391 @@
+        yield followThis;
+      }
+    }
+
+    iter these(param tag: iterKind, followThis) ref
+    where tag == iterKind.follower {
+      for i in privDom.these(tag, followThis) {
+        yield arr.dsiAccess[i];
+      }
+    }
+
+
+    //
+    // I/O
+    //
+
+    proc dsiSerialWrite(f) {
+      chpl_serialReadWriteRectangular(f, arr, privDom);
+    }
+
+    proc dsiSerialRead(f) {
+      chpl_serialReadWriteRectangular(f, arr, privDom);
+    }
+
+    proc dsiDisplayRepresentation() {
+      writeln("Slice view");
+      writeln("----------");
+      writeln("of domain:");
+      privDom.dsiDisplayRepresentation();
+      writeln("on array:");
+      arr.dsiDisplayRepresentation();
+      writeln("----------");
+    }
+
+
+    //
+    // accessors
+    //
+
+    inline proc dsiAccess(i: idxType ...rank) ref {
+      return dsiAccess(i);
+    }
+
+    inline proc dsiAccess(i: idxType ...rank)
+    where !shouldReturnRvalueByConstRef(eltType) {
+      return dsiAccess(i);
+    }
+
+    inline proc dsiAccess(i: idxType ...rank) const ref
+    where shouldReturnRvalueByConstRef(eltType) {
+      return dsiAccess(i);
+    }
+
+    inline proc dsiAccess(i) ref {
+      checkBounds(i);
+      if shouldUseIndexCache() {
+        const dataIdx = indexCache.getRADDataIndex(dom.stridable, i);
+        return indexCache.shiftedDataElem(dataIdx);
+      } else {
+        return arr.dsiAccess(i);
+      }
+    }
+
+    inline proc dsiAccess(i)
+    where !shouldReturnRvalueByConstRef(eltType) {
+      checkBounds(i);
+      if shouldUseIndexCache() {
+        const dataIdx = indexCache.getRADDataIndex(dom.stridable, i);
+        return indexCache.shiftedDataElem(dataIdx);
+      } else {
+        return arr.dsiAccess(i);
+      }
+    }
+
+    inline proc dsiAccess(i) const ref
+    where shouldReturnRvalueByConstRef(eltType) {
+      checkBounds(i);
+      if shouldUseIndexCache() {
+        const dataIdx = indexCache.getRADDataIndex(dom.stridable, i);
+        return indexCache.shiftedDataElem(dataIdx);
+      } else {
+        return arr.dsiAccess(i);
+      }
+    }
+
+    inline proc dsiLocalAccess(i) ref
+      return arr.dsiLocalAccess(i);
+
+    inline proc dsiLocalAccess(i)
+    where !shouldReturnRvalueByConstRef(eltType)
+      return arr.dsiLocalAccess(i);
+
+    inline proc dsiLocalAccess(i) const ref
+    where shouldReturnRvalueByConstRef(eltType)
+      return arr.dsiLocalAccess(i);
+
+    inline proc checkBounds(i) {
+      if boundsChecking then
+        if !privDom.dsiMember(i) then
+          halt("array index out of bounds: ", i);
+    }
+
+
+    //
+    // locality-oriented queries
+    //
+
+    proc dsiTargetLocales() {
+      return arr.dsiTargetLocales();
+    }
+
+    proc dsiHasSingleLocalSubdomain() param
+      return privDom.dsiHasSingleLocalSubdomain();
+
+    proc dsiLocalSubdomain() {
+      return privDom.dsiLocalSubdomain();
+    }
+
+    proc dsiNoFluffView() {
+      if canResolveMethod(arr, "dsiNoFluffView") {
```

Comment:
is this canResolveMethod here just to give a better error message? If that's the case, I'd like a comment to indicate that is the purpose (vs a workaround for a compiler bug).